spec/controllers/students_controller_spec.rb

```diff
@@ -0,0 +1,96 @@
+require 'rails_helper'
+
+RSpec.describe StudentsController, type: :controller do
+    #save instance of student in test db 
+    let!(:student) { create(:student, title: "Mr", first_name: "Harry", middle_name: "James", last_name: "Potter", email: "harry@test.com", gender: "Male", birth_date: "1989-07-23") }
+    
+    describe "GET #index" do
+        subject { get :index }
+
+        it "has a 200 status code" do
+            subject
+            expect(subject.status).to eq(200)
+        end
+        
+        it "renders the :index template" do
+            subject
+            expect(subject).to render_template("students/index")
+        end
+    
+        it "returns list of students" do
+            subject
+            expect(assigns(:students)).to eq [student]
+        end
+    end
+    
+    describe "GET #show" do
+        subject { get :show, id: student }
+
+        it "has a 200 status code" do 
+            subject
+            expect(response.status).to eq(200)
+        end
+        
+        it "renders the :show template" do
+            subject
+            expect(response).to render_template("students/show")
+        end
+
+        it "assigns requested student to @student" do
+            subject
+            expect(assigns(:student)).to eq student
+        end
+    end
+    
+    describe "GET #edit" do
+        subject { get :edit, id: student }
+        it "renders the :edit template" do
+            subject
+            expect(subject).to render_template("students/edit")
+        end
+        
+        it "assigned requested student to @student" do
+            subject
+            expect(assigns(:student)).to eq student
+        end
+        
+    end
+    
+    describe "PUT #update" do
+        context "with valid attributes" do
+            subject { put :update, id: student, student: attributes_for(:student, first_name: "Larry", last_name: "Potter", email: "harry@test.com", gender: "Male", birth_date: "1989-07-23") }
```

Comment:
Again .. your only checking the first_name attr below, which is fine.
so here you only need to actually change the first_name: 'Larry' attr, and the rest can be random.
```
subject { put :update, id: student, student: attributes_for(:student, first_name: "Larry")
```